backend/core/src/main/java/org/sonarsource/sonarlint/core/tracking/IssueMatchingService.java

```diff
@@ -140,6 +168,111 @@ public Map<Path, List<Either<ServerMatchedIssueDto, LocalOnlyIssueDto>>> trackWi
+            StreamSupport.stream(localMatchingResult.getUnmatchedRights().spliterator(), false).collect(Collectors.toList());
+          matchedAndUnmatchedIssues.addAll(unmatchedIssues);
+          return matchedAndUnmatchedIssues;
+        }));
+      updatedIssues.forEach((clientRelativePath, trackedIssues) -> storeTrackedIssues(knownIssuesStore, configurationScopeId, clientRelativePath, trackedIssues));
+      var issuesToRaise = getIssuesToRaise(updatedIssues);
+      client.raiseIssues(new RaiseIssuesParams(configurationScopeId, issuesToRaise, false, event.getAnalysisId()));
+      return;
+    }
+
+    var binding = effectiveBindingOpt.get();
+    var activeBranch = activeBranchOpt.get();
+    var translation = translationOpt.get();
+    if (event.isShouldFetchServerIssues()) {
+      var issuesByPath = rawIssuesByIdeRelativePath.entrySet().stream().collect(toMap(Map.Entry::getKey,
+        e -> e.getValue().stream().map(IssueMatchingService::toClientTrackedIssue).collect(toList())));
+      refreshServerIssues(new SonarLintCancelMonitor(), binding, activeBranch, issuesByPath, translation);
+    }
+    var newCodeDefinition = newCodeService.getFullNewCodeDefinition(configurationScopeId)
+      .orElse(NewCodeDefinition.withAlwaysNew());
+    newIssues = rawIssuesByIdeRelativePath.entrySet().stream().map(e -> {
+      var ideRelativePath = e.getKey();
+      var serverRelativePath = translation.ideToServerPath(ideRelativePath);
+      var serverIssues = storageService.binding(binding).findings().load(activeBranch, serverRelativePath);
+      var localOnlyIssues = localOnlyIssueStorageService.get().loadForFile(configurationScopeId, serverRelativePath);
+      var rawIssues = e.getValue();
+      var previouslyKnownIssues = knownIssuesStore.loadForFile(configurationScopeId, e.getKey());
+      List<TrackedIssue> allTrackedIssues;
+      if (previouslyKnownIssues != null && !previouslyKnownIssues.isEmpty()) {
+        var localIssueMatcher = new IssueMatcher<>(new KnownIssueMatchingAttributesMapper(), new RawIssueFindingMatchingAttributeMapper());
+        var localMatchingResult = localIssueMatcher.match(previouslyKnownIssues, rawIssues);
+        var newTrackedIssues = StreamSupport.stream(localMatchingResult.getUnmatchedRights().spliterator(), false)
+          .map(it -> toTrackedIssue(it, newCodeDefinition.isOnNewCode(Instant.now().toEpochMilli()))).collect(toSet());
+        var updatedMatchedIssues = localMatchingResult.getMatchedLefts().entrySet().stream()
+          .map(entry -> updateTrackedIssueWithRawIssueData(entry.getKey(), entry.getValue(), newCodeDefinition)).collect(toList());
+        allTrackedIssues = new ArrayList<>(newTrackedIssues);
+        allTrackedIssues.addAll(updatedMatchedIssues);
+      } else {
+        allTrackedIssues = rawIssues.stream()
+          .map(it -> toTrackedIssue(it, newCodeDefinition.isOnNewCode(Instant.now().toEpochMilli()))).collect(Collectors.toList());
+      }
```

Comment:
This part seems very similar to the standalone local issue matching. These two blocks can be unified.
In my opinion, basic steps of this method should be

```
match with local store
if(connectedMode) {
	match with server issues
	set new code true for unmatched issues
	set new code for server issues based on server NCD 
}
set introduction date for new issues
store to local
if(standalone mode) {
	set new code checking last 30 days
}
```

Reply:
I agree with you, let's see if you have time to tackle that today

Reply:
We will rework this part when we get rid of the things we kept for backward compatibility. See https://sonarsource.atlassian.net/browse/SLCORE-800